lib/bindings/http/unmarshaller.js

```diff
@@ -47,32 +39,39 @@ function resolveBindingName(payload, headers) {
   }
 }
 
-const Unmarshaller = function(receiverByBinding) {
-  this.receiverByBinding = receiverByBinding;
-};
+class Unmarshaller {
+  constructor(receiverByBinding) {
+    this.receiverByBinding = receiverByBinding;
+  }
 
-Unmarshaller.prototype.unmarshall = function(payload, headers) {
-  return new Promise((resolve, reject) => {
-    try {
-      validateArgs(payload, headers);
+  unmarshall(payload, headers) {
+    return new Promise((resolve, reject) => {
+      try {
+        if (!payload) {
+          throw new TypeError("payload is null or undefined");
+        }
+        if (!headers) {
+          throw new TypeError("headers is null or undefined");
+        }
```

Comment:
I like that you have removed the `validateArgs()` function, but I don't think there's actually a reason to throw here. A simpler version of what you have might look like this. 

```suggestion
  unmarshall(payload, headers) {
    return new Promise((resolve, reject) => {
      if (!payload) {
        return reject(new TypeError("payload is null or undefined"));
      }
      if (!headers) {
        return reject(new TypeError("payload is null or undefined"));
      }
```
I don't think the try/catch should be there at all, actually. Its only purpose is to catch this exception and then reject the promise. If something else, e.g. `resolveBindingName()` were to throw, the caller's catch block on the thennable will handle it.

Reply:
Thanks. I agree, but I'm just trying to refactor without modifying the functionality.
I propose we just approve this PR and can see about removing the existing try/catches here and elsewhere.

Reply:
OK - just to be clear though, this suggested modification isn't changing functionality. It's just a minor optimization similar to your removal of the `validateArgs()` function.

Reply:
OK, I can commit your suggested change but I think it's going to create a bad PR, where it was an OK PR before.